api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java

```diff
@@ -288,6 +288,44 @@ public void saveConcept_shouldForceSetFlagIfSetMembersExist() {
 		assertTrue(concept.getSet());
 	}
 	
+	/**
+	 * @see ConceptServiceImpl#saveDrug(Drug) 
+	 */
+	@Test
+	public void saveDrug_shouldPutGeneratedIdOntoReturnedDrug() {
+		executeDataSet("org/openmrs/api/include/ConceptServiceTest-initialConcepts.xml");
+		Drug drug = new Drug();
+		drug.setConcept(conceptService.getConcept(1));
+		Drug savedDrug = conceptService.saveDrug(drug);
+		assertNotNull(savedDrug.getDrugId());
+	}
+
+	/**
+	 * @see ConceptServiceImpl#saveDrug(Drug)
+	 */
+	@Test
+	public void saveDrug_shouldCreateNewDrugInDatabase() {
+		executeDataSet("org/openmrs/api/include/ConceptServiceTest-initialConcepts.xml");
+		Drug drug = new Drug();
+		drug.setConcept(conceptService.getConcept(1));
+		Drug savedDrug = conceptService.saveDrug(drug);
+		assertNotNull(conceptService.getDrug(savedDrug.getDrugId()));
```

Comment:
By this how did you able to ensure that it was created? actually this ID was retrieved through `conceptService.getConcept(1)`, which means it's already in the dataset. 

Can you create a pure Drug object using the properties, and check it for creation?